ARM Support

maturin is a dependency for orjson, which I'm hoping to use. However, there's no binary and it won't build. Here's an excerpt from `pip3 install maturin -vv`
```
Building wheels for collected packages: maturin
  Created temporary directory: /tmp/pip-wheel-xfduloq0
  Running setup.py bdist_wheel for maturin ...   Destination directory: /tmp/pip-wheel-xfduloq0
  Running command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-50kzifkp/maturin/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/pip-wheel-xfduloq0 --python-tag cp37
  running bdist_wheel
  running build
  installing to build/bdist.linux-armv7l/wheel
  running install
  error: [Errno 2] No such file or directory: 'cargo': 'cargo'
error
  Failed building wheel for maturin
  Running setup.py clean for maturin
  Running command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-50kzifkp/maturin/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" clean --all
  running clean
  'build/lib' does not exist -- can't clean it
  'build/bdist.linux-armv7l' does not exist -- can't clean it
  'build/scripts-3.7' does not exist -- can't clean it
Failed to build maturin
Installing collected packages: maturin
  Created temporary directory: /tmp/pip-record-qkx984lk
  Running setup.py install for maturin ...     Running command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-50kzifkp/maturin/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-qkx984lk/install-record.txt --single-version-externally-managed --compile
    running install
    error: [Errno 2] No such file or directory: 'cargo': 'cargo'
error
Cleaning up...
  Removing source in /tmp/pip-install-50kzifkp/maturin
Removed build tracker '/tmp/pip-req-tracker-v18k8qvo'
Command "/usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-50kzifkp/maturin/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-qkx984lk/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-install-50kzifkp/maturin/
Exception information:
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/pip/_internal/cli/base_command.py", line 143, in main
    status = self.run(options, args)
  File "/usr/lib/python3/dist-packages/pip/_internal/commands/install.py", line 386, in run
    use_user_site=options.use_user_site,
  File "/usr/lib/python3/dist-packages/pip/_internal/req/__init__.py", line 49, in install_given_reqs
    **kwargs
  File "/usr/lib/python3/dist-packages/pip/_internal/req/req_install.py", line 791, in install
    spinner=spinner,
  File "/usr/lib/python3/dist-packages/pip/_internal/utils/misc.py", line 723, in call_subprocess
    % (command_desc, proc.returncode, cwd))
pip._internal.exceptions.InstallationError: Command "/usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-50kzifkp/maturin/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-qkx984lk/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-install-50kzifkp/maturin/
```
Python 3.7.3
uname -r: `4.19.75-v7l+`

It was naive to assume it would just work - I hadn't installed cargo/rust.
Now it fails on compiling `ring`

```
error[E0658]: use of unstable library feature 'alloc': this library is unlikely to be stabilized in its current form or name (see issue #27783)
    --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.9/src/lib.rs:69:1
     |
  69 | extern crate alloc;
     | ^^^^^^^^^^^^^^^^^^^
```

--- a/setup.py
+++ b/setup.py
@@ -10,6 +10,7 @@
 # for maturin itself use `maturin sdist`.
 
 import os
+import shutil
 import subprocess
 import sys
 
@@ -48,6 +49,11 @@ class PostInstallCommand(install):
         if os.path.isfile(existing_binary):
             source = existing_binary
         else:
+            if not shutil.which("cargo"):
+                raise RuntimeError(
+                    "cargo not found in PATH. Please install rust "
+                    "(https://www.rust-lang.org/tools/install) and try again"
+                )
             subprocess.check_call(
                 ["cargo", "rustc", "--bin", "maturin", "--", "-C", "link-arg=-s"]
             )

